### ISSUE TITLE
VS2017 Find all references doesn't detect Object destructuring with accessors

### ISSUE THREAD

**Issue by danelkhen** [Microsoft/TypeScript#16922]
**TypeScript Version:**  2.4.1

**Code**

```ts
class Test {
    get testProp(): string {return "test";}
}
let obj: Test;
let { testProp } = obj;
```

**Expected behavior:**
Find all references on ```get testProp()``` should detect the last line ```let { testProp } = obj;``` as a reference (just the same way it works if ```testProp``` is a simple field or a method)

**Actual behavior:**
Find all references on ```get testProp()``` does not detect the last line ```let { testProp } = obj;``` as a reference

**Comment by andy-ms** [comment]
I can reproduce this:
```ts
/// <reference path="fourslash.ts" />

// @noLib: true

////class Test {
////    get [|x|]() { return 0; }
////}
////let obj: Test;
////let { [|x|] } = obj;

const [r0, r1] = test.ranges();
verify.referenceGroups(r0, []); //todo: fill in test
```

Running this test and putting a breakpoint at `const symbol =` in `getPropertyOfType` in `checker.ts` shows that we are getting back a `Transient` symbol with a mapper. The search symbol that we got from `checker.getSymbolAtLocation(node);` is a different non-transient symbol, so find-all-refs sees these as two separate symbols and thus not a match.
@sandersn Any idea why there would be a transient symbol here?

**Comment by sandersn** [comment]
If it works the same way as object rest/spread (I believe it does, but have not checked the spec), then `x` is transient because it isn't really an accessor any more. It's whatever the accessor returned. So *technically* they are not the same and it is correct for them to have different symbols. Consider `let { x: y } = obj;` where only the name `x` should change &mdash; the shortcut muddies things by combining them.

In the end, though, if we treat shortcut destructuring of simple properties as references to the property, then we should do the same for accessors.

**Comment by andy-ms** [comment]
To clarify, when we call `getSymbolAtLocation` we're at the declaration.
When we are searching for uses and come across a shorthand destructuring, we make sure to get the type of the binding pattern and call `getPropertyOfType`. So the problem is that the symbol at `get x() {}` is not the same as the result of `getPropertyOfType(Test, "x")`.
